### PR TITLE
Introduce workflow for stale PRs

### LINKED ISSUE #12796: Port PR management bot from Apache Beam
### Description

[Apache Beam](https://github.com/apache/beam) has a bot that labels incoming PRs, assigns reviewers, and comments if a PR is becoming stale ([dev list discussion](https://lists.apache.org/thread/v69hftt5nrnm4k35799g8jdhffv7j9fs)).
Maybe we can replicate some of the functionality for Lucene.

### PR DESCRIPTION
**Pull request by stefanvodita**
PRs get stale and we miss out on good contributions. This workflow will mark PRs that are becoming stale.

Addresses #12796

### CODE REVIEW

File: .github/workflows/stale.yml [hunk 1]
```diff
@@ -22,6 +22,7 @@ jobs:
       with:
         repo-token: ${{ secrets.GITHUB_TOKEN }}
         days-before-pr-stale: 14
+        exempt-draft-pr: true
```
Comment (uschindler) [inline .github/workflows/stale.yml:25]: thanks!

**Review by uschindler — APPROVED** [review]

**Comment by stefanvodita** [comment]
I ended up checking more of the configurable parameters after looking for the one to exclude draft PRs. Three things to note:
1. I have to configure a few more parameters to handle issues correctly and to not close PRs when they get too stale, which seems to be the default (scary!).
2. There is a `debug-only` mode we can enable, which will make it so the worflow runs without applying any changes. We can read its logs to see if it's behaving the way we want it to. If it is, we can disable debug mode and run it for real.
3. We can configure how many operations the workflow will do to avoid hitting the rate limit. The default is `30`, which is probably a good place to start. As a consequence, we would also avoid having most of the open PRs labeled stale in one go.

**Comment by stefanvodita** [comment]
I did some testing with the parameters I mentioned and pushed a new revision.
In the tests on my fork, a stale PR took 5 operations to process and a non-stale PR took 1.

**Review by uschindler — APPROVED** [review]

**Comment by mikemccand** [comment]
Thanks @uschindler and @stefanvodita -- the latest iteration looks great -- the `debug-only` flag means it won't actually make any changes, just go through the motions.

I'll merge now.

**Comment by stefanvodita** [comment]
Thanks Uwe for the review and Mike for merging.
The action logs look good. #12998 disables `debug-only` and increases the operations budget to 500. The total [per-hour budget is 1000](https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28#primary-rate-limit-for-github_token-in-github-actions) and I don't think we're using it up with the other workflows on the repo.

**Pull request merged**